test/e2e/resource_deletion_protection_test.go

```diff
@@ -0,0 +1,87 @@
+	}
+	noProtectedLabelValues := map[string]string{
+		workv1alpha2.DeletionProtectionLabelKey: "",
+	}
+
+	// create the deployment and namespaces for test.
+	ginkgo.BeforeEach(func() {
+		deploymentName = deploymentNamePrefix + rand.String(RandomStrLength)
+		namespaceName = "karmada-e2e-" + rand.String(RandomStrLength)
+		namespace = helper.NewNamespace(namespaceName)
+		deployment = helper.NewDeployment(namespaceName, deploymentName)
+		framework.CreateNamespace(kubeClient, namespace)
+		framework.CreateDeployment(kubeClient, deployment)
+	})
+
+	ginkgo.It("delete the protected deployment", func() {
+		framework.UpdateDeploymentLabels(kubeClient, deployment, protectedLabelValues)
+		err := kubeClient.AppsV1().Deployments(namespaceName).Delete(context.TODO(), deploymentName, metav1.DeleteOptions{})
+		// the deletion operation should return an error like:
+		// `admission webhook "resourcedeletionprotection.karmada.io" denied the request:
+		// This resource is protected, please make sure to remove the label:
+		// resourcetemplate.karmada.io/deletion-protected`
+		gomega.Expect(err).Should(gomega.HaveOccurred())
+	})
+
+	ginkgo.It("delete the protected namespace", func() {
+		framework.UpdateNamespaceLabels(kubeClient, namespace, protectedLabelValues)
+		err := kubeClient.CoreV1().Namespaces().Delete(context.TODO(), namespaceName, metav1.DeleteOptions{})
+		// the deletion operation should return an error too.
+		gomega.Expect(err).Should(gomega.HaveOccurred())
+	})
+
+	ginkgo.It("delete the no protected namespace, the deployment is protected", func() {
+		framework.UpdateDeploymentLabels(kubeClient, deployment, protectedLabelValues)
+		err := kubeClient.CoreV1().Namespaces().Delete(context.TODO(), namespaceName, metav1.DeleteOptions{})
+		// the deletion operation should not return an error, and the namespace
+		// should not be deleted because the deployment was protected.
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		_, err = kubeClient.CoreV1().Namespaces().Get(context.TODO(), namespaceName, metav1.GetOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	})
+
+	ginkgo.It("delete the namespace after the protection has been removed", func() {
+		framework.UpdateNamespaceLabels(kubeClient, namespace, protectedLabelValues)
+		err := kubeClient.CoreV1().Namespaces().Delete(context.TODO(), namespaceName, metav1.DeleteOptions{})
+		gomega.Expect(err).Should(gomega.HaveOccurred())
+
+		framework.UpdateNamespaceLabels(kubeClient, namespace, noProtectedLabelValues)
+		err = kubeClient.CoreV1().Namespaces().Delete(context.TODO(), namespaceName, metav1.DeleteOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	})
+
+	ginkgo.AfterEach(func() {
+		// remove the resource's protection.
+		framework.UpdateDeploymentLabels(kubeClient, deployment, noProtectedLabelValues)
+		framework.UpdateNamespaceLabels(kubeClient, namespace, noProtectedLabelValues)
```

Comment:
Here's the final cleanup action, I understand there should be no need to make updates to the namespace and deployment.

Reply:
the same as above